### PR TITLE
create-remote-secret: handle k8s 1.24+ sa token behavior

### PR DESCRIPTION
**Pull request by stevenctl**
A fix for https://github.com/istio/istio/issues/38246 that maintains current behavior. 

IIUC, this part doesn't break us:

> In 1.21+ secret-based tokens are no longer used for mounting into pods (ephemeral time-limited tokens are).

wip: need to add tests and release note

### CODE REVIEW

File: istioctl/pkg/multicluster/remote_secret.go [hunk 1]
```diff
@@ -238,6 +238,74 @@ func getServiceAccountSecret(client kube.ExtendedClient, opt RemoteSecretOptions
...
 	}
 
+	if !kube.IsAtLeastVersion(client, 24) {
+		return legacyGetServiceAccountSecret(serviceAccount, client, opt)
+	}
+	return getOrCreateServiceAccountSecret(serviceAccount, client, opt)
+
+}
+
+// In 1.24+ we can't assume the secrets will be referenced in the ServiceAccount or be created automatically.
```
Comment (nmittler) [inline istioctl/pkg/multicluster/remote_secret.go:248]: nit: In `Kubernetes` 1.24+...

File: istioctl/pkg/multicluster/remote_secret.go [hunk 2]
```diff
@@ -238,6 +238,74 @@ func getServiceAccountSecret(client kube.ExtendedClient, opt RemoteSecretOptions
...
+	ctx := context.TODO()
+
+	// manually specified secret, make sure it references the ServiceAccount
+	if opt.SecretName != "" {
+		secret, err := client.CoreV1().Secrets(opt.Namespace).Get(ctx, opt.SecretName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("could not get specified secret %s/%s: %v",
+				opt.Namespace, opt.SecretName, err)
+		}
+		if err := validateServiceAccountSecret(serviceAccount, secret); err != nil {
```
Comment (nmittler) [inline istioctl/pkg/multicluster/remote_secret.go:264]: Do we need something added to the analyzer for this as well?

**Review by nmittler — APPROVED** [review]

File: istioctl/pkg/multicluster/remote_secret.go (hunk 2 above)
Comment (stevenctl) [reply istioctl/pkg/multicluster/remote_secret.go:264]: I don't think so. 

1. We may not even use static tokens in some meshes. 
2. We really only read from this secret during remote-secret creation. The remote-secret won't have explicit references to this, just extracted content.

**Comment by Purushotham233** [comment]
@stevenctl 
sa token behaviour change is starting from 1.22. [Reference](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-an-api-token-for-a-serviceaccount)

Why the min. cluster version is configured as 24?

>  	if !kube.IsAtLeastVersion(client, 24) {
> 		return legacyGetServiceAccountSecret(serviceAccount, client, opt)
> 	}

**Pull request merged**